lua/ttt2/libraries/bodysearch.lua

```diff
@@ -819,7 +821,23 @@ if CLIENT then
 
     local function TypeToMaterial(type, data)
         if type == "wep" then
-            return util.WeaponForClass(data.wep).iconMaterial
+            -- in most cases the inflictor is a weapon and the weapon has a cached
+            -- material that can be used
+            local wep = util.WeaponForClass(data.wep).iconMaterial
+
+            if wep then
+                return wep.iconMaterial
```

Comment:
```suggestion
            local wep = util.WeaponForClass(data.wep)

            if wep then
                return wep.iconMaterial
```
umm.. Shouldnt that be like this or at least check weapon.iconMaterial?

Reply:
you are abolutely right, REALLY good catch. I only tested it on the boomerang of course, oof